src/nnet3/nnet-simple-component.cc

```diff
@@ -633,6 +633,135 @@ void NoOpComponent::Backprop(const std::string &debug_info,
   in_deriv->CopyFromMat(out_deriv);
 }
 
+// explicit
+SpatialRegularizationComponent::SpatialRegularizationComponent(
+    const SpatialRegularizationComponent &other):
+    dim_(other.dim_), regularization_scale_(other.regularization_scale_),
+    count_(other.count_), sumsq_(other.sumsq_) { }
+
+// virtual
+void SpatialRegularizationComponent::ZeroStats() {
+  sumsq_ = 0.0;
+  count_ = 0.0;
+}
+
+// virtual
+void SpatialRegularizationComponent::InitFromConfig(ConfigLine *cfl) {
+  int32 dim = 0;
+  bool ok = cfl->GetValue("dim", &dim);
+  BaseFloat regularization_scale = 0.1;
+  cfl->GetValue("regularization-scale", &regularization_scale);
+  if (!ok || cfl->HasUnusedValues() || regularization_scale < 0.0 || dim <= 0)
+    KALDI_ERR << "Invalid initializer for layer of type "
+              << Type() << ": \"" << cfl->WholeLine() << "\"";
+  Init(dim, regularization_scale);
+}
+
+
+void SpatialRegularizationComponent::Init(int32 dim, BaseFloat regularization_scale) {
+  KALDI_ASSERT(dim > 0 && regularization_scale >= 0.0);
+  dim_ = dim;
+  regularization_scale_ = regularization_scale_;
```

Comment:
[also from my code],
this should read
regularization_scale_ = regularization_scale;